### DISCUSSION TITLE
Misleading default value for `action`

### DISCUSSION THREAD

**Discussion by RogueOneEcho** [cross-seed/cross-seed#804] [General]
I've been testing out running `cross-seed` via docker with configuration by CLI options instead of `config.json`.

While doing so I've noted that a `config.json` is automatically created with values from `config.template.cjs`

What this means is that while the default value for some options such as `action` is listed as `save`

And that can be seen as the fallback value for the CLI here:

https://github.com/cross-seed/cross-seed.org/blob/44ba072acc4c566e1f708db9a35630bdc5803706/docs/basics/options.md?plain=1#L955

In reality the `config.json` is automatically populated with the value `inject`

So I suspect: 
1.  the default value in the docs should read `inject`
2. The fallback values defined in `cmd.ts` are actually redundant and it would make more sense to remove them and simply concatenate `config.template.cjs` to define the defaults to avoid the redundancy.

**Comment by mmgoodnow** [comment]
If nothing is listed in the config, the default action is save.

**Reply by RogueOneEcho** [reply]
I accidentally submitted that before I'd finished. My actual query was:

So I suspect: 
1.  the default value in the docs should read `inject`
2. The fallback values defined in `cmd.ts` are actually redundant and it would make more sense to remove them and simply concatenate `config.template.cjs` to define the defaults to avoid the redundancy.

**Reply by mmgoodnow** [reply]
It's intentionally set that way for backwards compatibility.

**Comment by zakkarry** [comment]
The current config template is for v6 on master, where as the documentation is still very much aimed at v5 during pre-release. We changed the default to inject for v6, previously it was save in v5 in the template.

**Reply by RogueOneEcho** [reply]
I suspected that might be the case, thanks for confirming

**Reply by zakkarry** [reply]
No problem, always good to have a reminder that I need to be aware of updating the options page once v6 is stable.

**Comment by RogueOneEcho** [comment]
@mmgoodnow @zakkarry  For v6 what would you consider the default value of options such as `rss-cadence`? 

`cmd.ts` does't list a fallback, but `config.template.cjs` defines it as `30 minutes` so when calling the CLI without a `config.json` it does default to `30 minutes`? 

https://github.com/cross-seed/cross-seed/blob/06c5f2b93f8d086df74ce503cbef51199dc6f21b/src/config.template.cjs#L339
https://github.com/cross-seed/cross-seed/blob/06c5f2b93f8d086df74ce503cbef51199dc6f21b/src/cmd.ts#L389-L392

I'm happy to update the default values in the docs to those defined in `config.template.cjs` but I understand why they're currently left blank.

**Reply by zakkarry** [reply]
The default is undefined if not provided. 

fallbacks do not indicate recommended settings, those are what we provide in the config.template.cjs

Standard recommendations to start with, where as fallbacks are if no value is provided.

**Reply by mmgoodnow** [reply]
If you want to put everything in CLI args, I'd use an empty config file.
